e2e/snapshot_e2e/lifecycle.py

```
@@ -7,8 +7,10 @@
 
 import shlex
 import time
-from typing import Any, Callable
+from contextlib import contextmanager
+from typing import Any, Callable, Iterator
```

Ruff (0.16.2)

[warning] Import from `collections.abc` instead: `Callable`, `Iterator`

Import from `collections.abc`

(UP035)